### DISCUSSION TITLE
请问一下端到端 halo 2 怎么用啊？

### DISCUSSION THREAD

**Discussion by voidcenter** [z2o-k7e/halo2-step-by-step#32] [Q&A] [answered]
目前看到的教程主要是在mock prover的环境下使用，如果想要用真的prover，然后在链上验证，这个工作流程是怎么样的呢？有没有类似于 snarkjs 这样的说明  https://github.com/iden3/snarkjs

**Comment by qizhou** [comment, marked as answer]
https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_proofs/examples/serialization.rs 这里有使用BN256的完整证明和验证过程。
